programs/drift/src/controller/position.rs

```diff
@@ -384,42 +381,22 @@ pub fn update_lp_market_position(
     fee_to_market: i128,
     liquidity_split: AMMLiquiditySplit,
 ) -> DriftResult<(i128, i128, i128)> {
-    let user_lp_shares = market.amm.user_lp_shares;
-
-    if user_lp_shares == 0 || liquidity_split == AMMLiquiditySplit::ProtocolOwned {
+    if market.amm.user_lp_shares == 0 || liquidity_split == AMMLiquiditySplit::ProtocolOwned {
         return Ok((0, 0, 0)); // no need to split with LP
     }
 
-    let total_lp_shares = if liquidity_split == AMMLiquiditySplit::LPOwned {
-        market.amm.user_lp_shares
-    } else {
-        market.amm.sqrt_k
-    };
-
-    // update Market per lp position
-    let per_lp_delta_base = get_proportion_i128(
-        delta.base_asset_amount.cast()?,
-        AMM_RESERVE_PRECISION,
-        total_lp_shares,
-    )?;
+    let (per_lp_delta_base, per_lp_delta_quote, per_lp_fee) =
+        market
+            .amm
+            .calculate_per_lp_delta(delta, fee_to_market, liquidity_split)?;
 
-    let mut per_lp_delta_quote = get_proportion_i128(
-        delta.quote_asset_amount.cast()?,
-        AMM_RESERVE_PRECISION,
-        total_lp_shares,
+    let (lp_delta_base, lp_delta_quote, lp_fee) = market.amm.calculate_lp_delta(
```

Comment:
`lp_delta_quote` and `lp_fee` don't seem used? could remove and save some compute units?